pycyphal/transport/udp/_session/_input.py

```diff
@@ -97,61 +70,115 @@ def __init__(
+                else:
+                    ts, frame = self._frame_queue.get_nowait()
+            except (asyncio.TimeoutError, asyncio.QueueEmpty):
+                # If there are unprocessed transfers, allow the caller to read them even if the instance is closed.
+                if self._finalizer is None:
+                    raise pycyphal.transport.ResourceClosedError(f"{self} is closed") from None
+                return None
+            if frame is None:
+                self._statistics.errors += 1
+                continue
+            # это проблема но мы это потом починим
+            if frame.data_specifier != self._specifier.data_specifier:
+                continue
+            if frame.source_node_id == self._local_node_id:
+                continue
+            if not self.specifier.is_promiscuous:
+                if frame.source_node_id != self.specifier.remote_node_id:
+                    continue
+            self._statistics.frames += 1
+            source_node_id = frame.source_node_id
+            assert (
+                isinstance(source_node_id, int) and 0 <= source_node_id <= NODE_ID_MASK
+            ), "Internal protocol violation"
+            # Anonymous - no reconstruction needed
+            if source_node_id == NODE_ID_MASK:
+                transfer = TransferReassembler.construct_anonymous_transfer(ts, frame)
             else:
-                transfer = self._queue.get_nowait()
-        except (asyncio.TimeoutError, asyncio.QueueEmpty):
-            # If there are unprocessed transfers, allow the caller to read them even if the instance is closed.
-            if self._maybe_finalizer is None:
-                raise pycyphal.transport.ResourceClosedError(f"{self} is closed") from None
-            return None
-        else:
-            assert isinstance(transfer, pycyphal.transport.TransferFrom), "Internal protocol violation"
-            assert transfer.source_node_id == self._specifier.remote_node_id or self._specifier.remote_node_id is None
-            return transfer
+                _logger.debug("%s: Processing frame %s", self, frame)
+                transfer = self._get_reassembler(source_node_id).process_frame(ts, frame, self._transfer_id_timeout)
+            if transfer is not None:
+                self._statistics.transfers += 1
+                self._statistics.payload_bytes += sum(map(len, transfer.fragmented_payload))
+                _logger.debug("%s: Received transfer %s; current stats: %s", self, transfer, self._statistics)
+                return transfer
+
+    def put_into_queue(self, ts: pycyphal.transport.Timestamp, frame: typing.Optional[UDPFrame]) -> None:
+        self._frame_queue.put_nowait((ts, frame))
+
+    def _reader_thread(self, loop: asyncio.AbstractEventLoop) -> None:
+        while not self._closed and self._sock.fileno() >= 0:
+            try:
+                read_ready, _, _ = select.select([self._sock], [], [], 0.1)
```

Comment:
```suggestion
                # TODO: add a dedicated socket for aborting the select call
                # when self.close() is invoked to avoid blocking on
                # self._thread.join() in self.close().
                read_ready, _, _ = select.select([self._sock], [], [], 0.1)
```